src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rest/QueryRcaRequestHandler.java

```diff
@@ -162,13 +162,20 @@ public void handle(HttpExchange exchange) throws IOException {
 
   private void handleClusterRcaRequest(Map<String, String> params, HttpExchange exchange)
       throws IOException {
+    //check if we are querying from elected master
+    if (!validNodeRole()) {
+      JsonObject errResponse = new JsonObject();
+      errResponse.addProperty("error", "Node being queried is not elected master.");
+      sendResponse(exchange, errResponse.toString(),
+              HttpURLConnection.HTTP_BAD_REQUEST);
+      return;
+    }
     List<String> rcaList = metricsRestUtil.parseArrayParam(params, NAME_PARAM, true);
     // query all cluster level RCAs if no RCA is specified in name.
     if (rcaList.isEmpty()) {
-      rcaList = SQLiteQueryUtils.getClusterLevelRca();
-    }
-    //check if RCA is valid
-    if (!validParams(rcaList)) {
+      // rcaList = SQLiteQueryUtils.getClusterLevelRca();
+      rcaList = persistable.getAllPersistedRcas();
```

Comment:
this will contain RCAs that are not cluster level RCAs. Do we also want to expose node level RCAs on local master node to user ?

Reply:
I would like to. Eventually, We would want to expose the RCAs on the local node, if they are not elected master.

The only thing with RCAs are that they are nested so, we won't want one node to gather RCAs from all nodes but if the request hits a node, it should be able to get all local RCAs.